compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1159,6 +1158,7 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
         srcOffHigh >= 0 && dstOffHigh >= 0 && copyLenHigh >= 0)
       {
       transformTheCall = true;
```

Comment:
Did you mean to use OR instead of AND there? I thought of that as well. This change keeps the current behavior in that if primitive **or** reference array are supported and we have no type information then we carry out the transformation anyway.

I could not figure out why this was even correct to begin with. In my opinion setting this value to true by default and figuring out based on information we know whether to set it to false is too aggressive. We should be setting it to false by default and figuring out based on the information we know whether it should be true.

This is a much less aggressive approach but also more risky and probably outside the scope of this change. Do any of the reviewers have other thoughts?